### PR TITLE
update some gem dependencies

### PR DESCRIPTION
**Pull request by weiweishi**
A much-needed gem upgrade. Some features are not working properly because of the outdated gems: i.e.:  customized logo etc with Dragonfly, possible due to this: https://stackoverflow.com/questions/42538904/rails-dragonfly-uid-not-found-although-it-displays-the-path-correctly. 

Unfortunately, we don't have a proper testing env set up for DMP Assistance. We were hoping to revamp this project with the new codebase upgrade. So for now, we'd have to deploy this into staging to manually test the change.

### CODE REVIEW

**Comment by pgwillia** [comment]
A couple things.  
- requires Ruby 2.3 so README could be updated too
- getting an error message that isn't in master
```
/home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/railties-3.2.22.5/lib/rails/railtie/configuration.rb:85:in `method_missing': undefined method `assets' for #<Rails::Engine::Configuration:0x00005575ed2e87f0> (NoMethodError)
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/bourbon-5.1.0/lib/bourbon.rb:7:in `<class:Engine>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/bourbon-5.1.0/lib/bourbon.rb:6:in `<module:Bourbon>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/bourbon-5.1.0/lib/bourbon.rb:4:in `<top (required)>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `block in require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:236:in `load_dependency'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activeadmin-1.0.0.pre1/lib/active_admin.rb:6:in `<top (required)>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `block in require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:236:in `load_dependency'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activeadmin-1.0.0.pre1/lib/activeadmin.rb:1:in `<top (required)>'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:81:in `require'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:81:in `block (2 levels) in require'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:76:in `each'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:76:in `block in require'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:65:in `each'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:65:in `require'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler.rb:114:in `require'
	from /home/pjenkins/Code/DMPonline_v4/config/application.rb:8:in `<top (required)>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/railties-3.2.22.5/lib/rails/commands.rb:53:in `require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/railties-3.2.22.5/lib/rails/commands.rb:53:in `block in <top (required)>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/railties-3.2.22.5/lib/rails/commands.rb:50:in `tap'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/railties-3.2.22.5/lib/rails/commands.rb:50:in `<top (required)>'
	from script/rails:6:in `require'
	from script/rails:6:in `<main>'
```
something up with https://github.com/ualbertalib/DMPonline_v4/blob/da79f8e1e04d71fc32ba39b8a169341eb61dc17c/config/application.rb#L6-L11

**Comment by weiweishi** [comment]
@pgwillia could you please let me know when you run into this particular issue? Did it happen during `bundle update`? (also, the code is actually not on the master branch - early on we had things set to run off the `deployment-portage` branch. )

**Comment by pgwillia** [comment]
> @pgwillia could you please let me know when you run into this particular issue? Did it happen during `bundle update`? (also, the code is actually not on the master branch - early on we had things set to run off the `deployment-portage` branch. )

Right.  My mistake by master branch I meant `deployment-portage`.  Running the rails server in this branch works for me `bundle exec rails s`.
Screenshot from 2019-05-21 10-38-14

But after `bundle install`  on the `gem-update` branch running the rails server results in:
```
pjenkins@DIG002TJ1:~/Code/DMPonline_v4$ bundle exec rails s
/home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/railties-3.2.22.5/lib/rails/railtie/configuration.rb:85:in `method_missing': undefined method `assets' for #<Rails::Engine::Configuration:0x0000557ad6ac5ef8> (NoMethodError)
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/bourbon-5.1.0/lib/bourbon.rb:7:in `<class:Engine>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/bourbon-5.1.0/lib/bourbon.rb:6:in `<module:Bourbon>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/bourbon-5.1.0/lib/bourbon.rb:4:in `<top (required)>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `block in require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:236:in `load_dependency'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activeadmin-1.0.0.pre1/lib/active_admin.rb:6:in `<top (required)>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `block in require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:236:in `load_dependency'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activesupport-3.2.22.5/lib/active_support/dependencies.rb:251:in `require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/activeadmin-1.0.0.pre1/lib/activeadmin.rb:1:in `<top (required)>'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:81:in `require'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:81:in `block (2 levels) in require'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:76:in `each'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:76:in `block in require'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:65:in `each'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler/runtime.rb:65:in `require'
	from /home/pjenkins/.rvm/rubies/ruby-2.3.8/lib/ruby/site_ruby/2.3.0/bundler.rb:114:in `require'
	from /home/pjenkins/Code/DMPonline_v4/config/application.rb:8:in `<top (required)>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/railties-3.2.22.5/lib/rails/commands.rb:53:in `require'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/railties-3.2.22.5/lib/rails/commands.rb:53:in `block in <top (required)>'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/railties-3.2.22.5/lib/rails/commands.rb:50:in `tap'
	from /home/pjenkins/Code/DMPonline_v4/vendor/ruby/2.3.0/gems/railties-3.2.22.5/lib/rails/commands.rb:50:in `<top (required)>'
	from script/rails:6:in `require'
	from script/rails:6:in `<main>'
```

Same same after `bundle update`

**Comment by pgwillia** [comment]
The offending gem is `bourbon`

**Comment by weiweishi** [comment]
was able to replicate this issue on `deployment-portage' branch as well...
```
bundler-1.16.4/lib/bundler/runtime.rb:84:in `rescue in block (2 levels) in require': There was an error while trying to load the gem 'activeadmin'. (Bundler::GemRequireError)
Gem Load Error is: undefined method `assets' for #<Rails::Engine::Configuration:0x00558e830ac4d8>
```

File: Gemfile [hunk 1]
```diff
@@ -86,7 +95,8 @@ gem 'htmltoword'
 group :development, :test do
   gem 'rspec-rails'
   gem 'rspec'
-  gem 'selenium-webdriver'
+  # pinned this version to work with Ruby 2.1 and Rails 3
+  gem 'selenium-webdriver', '=3.141.0'
```
Comment (pgwillia) [inline Gemfile]:
```suggestion
  gem 'selenium-webdriver', '3.141.0'
```

`=` is the default. https://www.devalot.com/articles/2012/04/gem-versions.html#available-operators What do you think about leaving this out?

File: Gemfile [hunk 2]
```diff
@@ -5,10 +5,19 @@ gem 'rails', '~>3.2.22'
 # pin this to post-CVE 2017-5946
 gem 'rubyzip', '~> 1.2.2'
 
+# pin the following gems to work with Ruby 2.1 and Rails 3
+gem 'arbre','~>1.1.1'
```
Comment (pgwillia) [inline Gemfile]:
```suggestion
gem 'arbre','1.1.1'
```

File: Gemfile [hunk 3]
```diff
@@ -5,10 +5,19 @@ gem 'rails', '~>3.2.22'
...
+gem 'rb-inotify','~>0.9.10'
```
Comment (pgwillia) [inline Gemfile]:
```suggestion
gem 'rb-inotify','0.9.10'
```

File: Gemfile [hunk 4]
```diff
@@ -5,10 +5,19 @@ gem 'rails', '~>3.2.22'
...
+gem 'public_suffix','~>3.0.3'
```
Comment (pgwillia) [inline Gemfile]:
```suggestion
gem 'public_suffix','3.0.3'
```

File: Gemfile [hunk 5]
```diff
@@ -5,10 +5,19 @@ gem 'rails', '~>3.2.22'
...
+gem 'bourbon','~>4.2.7'
```
Comment (pgwillia) [inline Gemfile]:
```suggestion
gem 'bourbon','4.2.7'
```

File: Gemfile [hunk 6]
```diff
@@ -5,10 +5,19 @@ gem 'rails', '~>3.2.22'
...
+
 gem 'mysql2', '~>0.3.20'
 gem 'omniauth'
 gem 'omniauth-shibboleth'
-gem 'recaptcha'
+
+# pin recaptcha to work with Ruby 2.1 and Rails 3
+# some work is required to upgrade to 4.9. So pinned it to 3.x for now
+gem 'recaptcha', '~>3.4.0'
```
Comment (pgwillia) [inline Gemfile]:
```suggestion
gem 'recaptcha', '3.4.0'
```

File: Gemfile (hunk 1 above)
Comment (weiweishi) [reply Gemfile]: @pgwillia I plan to close this PR if all the updates are handled individually with dependabot or separate PRs. Many changes included here have already been addressed elsewhere. Also, I agree, we should leave this one out.

**Review by pgwillia — CHANGES_REQUESTED** [review]
If you like my suggestions we'll also need to update the Gemfile.lock.

File: Gemfile (hunk 1 above)
Comment (pgwillia) [reply Gemfile]: Gotcha!  The case of too many open pull requests ;)

File: Gemfile.lock [hunk 7]
```diff
@@ -201,54 +201,59 @@ GEM
...
-    rspec-support (3.5.0)
+      rspec-core (~> 3.8.0)
+      rspec-expectations (~> 3.8.0)
+      rspec-mocks (~> 3.8.0)
+      rspec-support (~> 3.8.0)
+    rspec-support (3.8.0)
     ruby-rc4 (0.1.5)
     rubyzip (1.2.3)
-    sass (3.4.22)
+    sass (3.7.4)
```
Comment (pgwillia) [inline Gemfile.lock:245]: sass is deprecated.  Could use sassc instead.

```
Ruby Sass has reached end-of-life and should no longer be used.

* If you use Sass as a command-line tool, we recommend using Dart Sass, the new
  primary implementation: https://sass-lang.com/install

* If you use Sass as a plug-in for a Ruby web framework, we recommend using the
  sassc gem: https://github.com/sass/sassc-ruby#readme

* For more details, please refer to the Sass blog:
  https://sass-lang.com/blog/posts/7828841
```

File: Gemfile.lock [hunk 8]
```diff
@@ -201,54 +201,59 @@ GEM
...
+      sass-listen (~> 4.0.0)
+    sass-listen (4.0.0)
+      rb-fsevent (~> 0.9, >= 0.9.4)
+      rb-inotify (~> 0.9, >= 0.9.7)
     sass-rails (3.2.6)
       railties (~> 3.2.0)
       sass (>= 3.1.10)
       tilt (~> 1.3)
-    selenium-webdriver (2.53.4)
+    selenium-webdriver (3.141.0)
```
Comment (pgwillia) [inline Gemfile.lock]: The tests need sorting out.  Not sure if this bump (rspec too) helps or hurts.

File: Gemfile.lock [hunk 9]
```diff
@@ -258,14 +263,14 @@ GEM
...
+    thin (1.7.2)
       daemons (~> 1.0, >= 1.0.9)
       eventmachine (~> 1.0, >= 1.0.4)
       rack (>= 1, < 3)
     thor (0.20.3)
-    thread_safe (0.3.5)
+    thread_safe (0.3.6)
     tilt (1.4.1)
-    tinymce-rails (4.4.1)
+    tinymce-rails (5.0.5)
```
Comment (pgwillia) [inline Gemfile.lock:274]: I looked at https://www.tiny.cloud/docs/migration-from-4x/  and didn't see anything that was impacted.

**Review by pgwillia — DISMISSED** [review]

File: Gemfile.lock (hunk 7 above)
Comment (weiweishi) [reply Gemfile.lock:245]: @pgwillia good idea. Will open a separate ticket for this change.

File: Gemfile.lock (hunk 8 above)
Comment (weiweishi) [reply Gemfile.lock]: I will downgrade this to current version, and we can figure this out separately

**Review by pgwillia — APPROVED** [review]
👍🎉

**Pull request merged**